scanner/src/test/kotlin/scanners/fossid/FossIdTest.kt

```diff
@@ -0,0 +1,829 @@
+            val fossId = createFossId(config)
+
+            fossId.version shouldBe ""
+        }
+    }
+
+    "scanPackages" should {
+        "create a new scan for an existing project" {
+            val projectCode = projectCode(PROJECT)
+            val scanCode = scanCode(PROJECT, null)
+            val config = createConfig(deltaScans = false)
+            val vcsInfo = createVcsInfo()
+            val scan = createScan(vcsInfo.url, "${vcsInfo.revision}_other", scanCode)
+
+            val service = config.createService()
+                .expectProjectRequest(projectCode)
+                .expectListScans(projectCode, listOf(scan))
+                .expectCheckScanStatus(scanCode, ScanState.FINISHED)
+                .expectCreateScan(projectCode, scanCode, vcsInfo)
+                .expectDownload(scanCode)
+                .mockFiles(scanCode, identifiedRange = 1..2, markedRange = 1..2)
+
+            val fossId = createFossId(config)
+
+            fossId.scan(listOf(createPackage(createIdentifier(index = 1), vcsInfo)))
+
+            coVerify {
+                service.createScan(USER, API_KEY, projectCode, scanCode, vcsInfo.url, vcsInfo.revision)
+                service.downloadFromGit(USER, API_KEY, scanCode)
+                service.checkDownloadStatus(USER, API_KEY, scanCode)
+            }
+
+            coVerify(exactly = 0) {
+                service.createProject(any())
+            }
+        }
+
+        "return copyright and license findings from a new scan" {
+            val projectCode = projectCode(PROJECT)
+            val scanCode = scanCode(PROJECT, null)
+            val config = createConfig(deltaScans = false)
+            val vcsInfo = createVcsInfo()
+            val scan = createScan(vcsInfo.url, "${vcsInfo.revision}_other", scanCode)
+            val pkgId = createIdentifier(index = 42)
+
+            config.createService()
+                .expectProjectRequest(projectCode)
+                .expectListScans(projectCode, listOf(scan))
+                .expectCheckScanStatus(scanCode, ScanState.FINISHED)
+                .expectCreateScan(projectCode, scanCode, vcsInfo)
+                .expectDownload(scanCode)
+                .mockFiles(scanCode, identifiedRange = 1..2, markedRange = 1..2)
+
+            val fossId = createFossId(config)
+
+            val summary = fossId.scan(listOf(createPackage(pkgId, vcsInfo))).summary(pkgId)
+
+            val expectedCopyrightFindings = listOf(
+                CopyrightFinding("copyrightMarked1", textLocation(1)),
+                CopyrightFinding("copyrightMarked2", textLocation(2))
+            )
+            val expectedLicenseFindings = listOf(
+                LicenseFinding("licenseMarkedIdentifier1", textLocation(1)),
+                LicenseFinding("licenseMarkedIdentifier2", textLocation(2))
+            )
+            summary.copyrightFindings shouldContainExactlyInAnyOrder expectedCopyrightFindings
```

Comment:
I have a feeling you are mixing two kind of tests in one file: the tests with `coVerify` where you verify if a function was called, and the test on `summary` when you test the data returned by the scanner.
Shouldn't you split this test in two files, according the test type ? The added benefit would be that this file would not be so big.

Reply:
I don't see this as different tests. Both call the mock service; these are just different variants to test the interactions. The tests use common infrastructure to setup the scenario and the mock objects. If you split them, you either have to duplicate this logic or extract it somehow into shared test helper objects.

Reply:
Yes, I would do the latter.
However, since this is your test, you can leave it as it is.